python/mxnet/gluon/utils.py

```diff
@@ -207,4 +207,14 @@ def download(url, path=None, overwrite=False, sha1_hash=None):
                 if chunk: # filter out keep-alive new chunks
                     f.write(chunk)
 
+        if sha1_hash and not check_sha1(fname, sha1_hash):
+            raise UserWarning('File {} is downloaded but the content hash does not match. ' \
```

Comment:
delete the downloaded file when hash doesn't match?

Reply:
It will re-download next time. Also, deleting the file deprives the user of the chance to examine the file to find problems.

Reply:
even without overwrite=True?

Reply:
yes, read the code

Reply:
i see